src/components/media_manager/test/media_manager_impl_test.cc

```diff
@@ -33,147 +33,364 @@
+const uint32_t kConnectionKey = 1u;
+
+void dealloc_data(uint8_t* data) {
+  delete[] data;
+}
+
+void dealloc_file(std::ofstream* test_file) {
+  delete test_file;
+}
+}  // namespace
+
+typedef NiceMock<application_manager_test::MockApplication> MockApp;
+typedef SharedPtr<MockApp> MockAppPtr;
+typedef SharedPtr<MockMediaAdapterImpl> MockMediaAdapterImplPtr;
 
 class MediaManagerImplTest : public ::testing::Test {
+ public:
+  // media_adapter_mock_ will be deleted in media_manager_impl (dtor)
+  MediaManagerImplTest() : media_adapter_mock_(new MockMediaAdapter()) {
+    media_adapter_listener_mock_ =
+        utils::MakeShared<MockMediaAdapterListener>();
+    ON_CALL(mock_media_manager_settings_, video_server_type())
+        .WillByDefault(ReturnRef(kDefaultValue));
+    ON_CALL(mock_media_manager_settings_, audio_server_type())
+        .WillByDefault(ReturnRef(kDefaultValue));
+    mock_app_ = ::utils::MakeShared<MockApp>();
+    media_manager_impl_.reset(
+        new MediaManagerImpl(app_mngr_, mock_media_manager_settings_));
+  }
+
  protected:
+  void StartMicrophoneCheckHelper() {
+    EXPECT_CALL(app_mngr_, application(kApplicationKey))
+        .WillOnce(Return(mock_app_));
+    EXPECT_CALL(mock_media_manager_settings_, app_storage_folder())
+        .WillOnce(ReturnRef(kStorageFolder));
+    EXPECT_CALL(mock_media_manager_settings_, app_resource_folder())
+        .WillOnce(ReturnRef(kResourceFolder));
+    EXPECT_CALL(mock_media_manager_settings_, recording_file_source())
+        .WillOnce(ReturnRef(kRecordingFileSource));
+  }
+
+  void InitMediaManager(const std::string& server_type) {
+    EXPECT_CALL(mock_media_manager_settings_, video_server_type())
+        .WillRepeatedly(ReturnRef(server_type));
+    EXPECT_CALL(mock_media_manager_settings_, audio_server_type())
+        .WillRepeatedly(ReturnRef(server_type));
+    if (server_type == kSocketValue) {
+      EXPECT_CALL(mock_media_manager_settings_, server_address())
+          .WillRepeatedly(ReturnRef(kServer_address));
+      EXPECT_CALL(mock_media_manager_settings_, video_streaming_port())
+          .WillOnce(Return(kVideoStreamingPort));
+      EXPECT_CALL(mock_media_manager_settings_, audio_streaming_port())
+          .WillOnce(Return(kAudioStreamingPort));
+    } else if (server_type == kPipeValue) {
+      EXPECT_CALL(mock_media_manager_settings_, named_video_pipe_path())
+          .WillOnce(ReturnRef(kNamedAudioPipePath));
+      EXPECT_CALL(mock_media_manager_settings_, named_audio_pipe_path())
+          .WillOnce(ReturnRef(kNamedAudioPipePath));
+    } else if (server_type == kFileValue) {
+      EXPECT_CALL(mock_media_manager_settings_, video_stream_file())
+          .WillOnce(ReturnRef(kVideoStreamFile));
+      EXPECT_CALL(mock_media_manager_settings_, audio_stream_file())
+          .WillOnce(ReturnRef(kAudioStreamFile));
+    }
+    if ((server_type == kPipeValue) || (server_type == kFileValue)) {
```

Comment:
@indlu , not sure about these `if ... else` in test. Good practice to NOT use conditions in tests